src/test/java/bio/overture/ego/utils/RedirectsTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2019. The Ontario Institute for Cancer Research. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package bio.overture.ego.utils;
+
+import static bio.overture.ego.utils.Redirects.*;
+import static org.junit.Assert.*;
+
+import bio.overture.ego.model.entity.Application;
+import lombok.val;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.springframework.security.oauth2.common.exceptions.UnauthorizedClientException;
+
+public class RedirectsTest {
```

Comment:
should we have a test case with url-encoded query params? something like `http://example.com/path?query=123`

Reply:
yes, I'll add

Reply:
but with the current matching strategy (host+port) this won't guarantee you will get the correct path.. if you have: `http://example.com/path?query=123` & `http://example.com/path2`

Reply:
What do you mean? If you provide a redirect uri from the client, any registered uri just needs to match the authority(domain, port) for the provided uri to be accepted

Reply:
yeah my bad, I just confused myself

Reply:
:spaghetti: no worries, it is confusing as its supporting two modes of operation at the same time :spaghetti: